src/main/webapp/js/captions.js

```diff
@@ -180,18 +180,28 @@ function beginCaptionRequest(videoId, url) {
     gapi.client.youtube.captions.list({
       "videoId": videoId,
       "part": [
-        "id"
+        "snippet"
       ]
     }).then(function(response) {
         if (response.result != null & response.result.items != null &
             response.result.items.length > 0) {
-
-            const trackId = response.result.items[0].id;
-
-            getCaptions(trackId, url).then(json => {
-                // send to backend
-                sendJsonForm(json);
-            });
+            
+            var i;
+            var trackId = "";
+            for (i = 0; i < response.result.items.length; i++) {
+                if (response.result.items[i].snippet.language === "en") {
+                    trackId = response.result.items[i].id;
```

Comment:
should we just `break` from the loop once we found the English one?

Reply:
done